must-gather/collection-scripts/gather_crds

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+source pwait
+max_parallelism=10
+
+# Cluster passed in from main gather
+clusterID=$1
+
+# Resource list
+resources=()
+
+# OADP
+resources+=($(/usr/bin/oc get crd | awk '/oadp.openshift.io/{print $1}'))
+
+# Velero
+resources+=($(/usr/bin/oc get crd | awk '/velero.io/{print $1}'))
+
+echo "Starting collection of CRDs: [${resources[@]}]"
+object_collection_path=/must-gather/clusters/${clusterID}/cluster-scoped-resources/apiextensions.k8s.io/customresourcedefinitions
```

Comment:
Should this be with quotes surrounding the var?

Reply:
The quotes problem could be verified with shellcheck (a shell script linter). As a suggestion, would add one more task in must-gather refactor to add shellcheck to project CI and target must-gather scripts, instead of fixing them now. What you think?

To test shellcheck in `must-gather/collection-scripts/gather_crs` and `must-gather/collection-scripts/gather_crds`, run
```sh
docker run --rm -v "$PWD:/mnt" koalaman/shellcheck:stable must-gather/collection-scripts/gather_crs
docker run --rm -v "$PWD:/mnt" koalaman/shellcheck:stable must-gather/collection-scripts/gather_crds
```

Reply:
For this script to make it proper I would add double quotes (It's not refactor problem, but the implementation problem). Refactor does change the way we write things, but it's not resolving bugs, here we have potential problem that may lead to bugs.

Running shellcheck is always great for those types of scripts and actually it pointed out some other errors (note the warnings/info can be ignored, but errors should be resolved).

```
$ shellcheck -x gather_crs 

In gather_crs line 15:
  resources+=($i)
              ^-- SC2206 (warning): Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.


In gather_crs line 20:
  resources+=($i)
              ^-- SC2206 (warning): Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.


In gather_crs line 23:
echo "Starting collection of: [${resources[@]}]"
                               ^-------------^ SC2145 (error): Argument mixes string and array. Use * or separate argument.


In gather_crs line 26:
for resource in ${resources[@]}; do
                ^-------------^ SC2068 (error): Double quote array expansions to avoid re-splitting elements.


In gather_crs line 28:
  /usr/bin/oc get ${resource} --all-namespaces -o custom-columns=NAME:.metadata.name,NAMESPACE:.metadata.namespace --no-headers 2> /dev/null | \
                  ^---------^ SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
  /usr/bin/oc get "${resource}" --all-namespaces -o custom-columns=NAME:.metadata.name,NAMESPACE:.metadata.namespace --no-headers 2> /dev/null | \


In gather_crs line 29:
  while read ocresource; do
        ^--^ SC2162 (info): read without -r will mangle backslashes.


In gather_crs line 30:
    ocobject=$(echo $ocresource | awk '{print $1}')
                    ^---------^ SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
    ocobject=$(echo "$ocresource" | awk '{print $1}')


In gather_crs line 31:
    ocproject=$(echo $ocresource | awk '{print $2}')
                     ^---------^ SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
    ocproject=$(echo "$ocresource" | awk '{print $2}')


In gather_crs line 32:
    if [ -z "${ocproject}" ]|[ "${ocproject}" == "<none>" ]; then
                            ^-- SC2266 (warning): Use || for logical OR. Single | will pipe.

Did you mean: 
    if [ -z "${ocproject}" ]||[ "${ocproject}" == "<none>" ]; then


In gather_crs line 34:
      mkdir -p ${object_collection_path}
               ^-----------------------^ SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
      mkdir -p "${object_collection_path}"


In gather_crs line 35:
      /usr/bin/oc get ${resource} -o yaml ${ocobject} &> ${object_collection_path}/${ocobject}.yaml &
                      ^---------^ SC2086 (info): Double quote to prevent globbing and word splitting.
                                          ^---------^ SC2086 (info): Double quote to prevent globbing and word splitting.
                                                         ^-----------------------^ SC2086 (info): Double quote to prevent globbing and word splitting.
                                                                                   ^---------^ SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
      /usr/bin/oc get "${resource}" -o yaml "${ocobject}" &> "${object_collection_path}"/"${ocobject}".yaml &


In gather_crs line 39:
      if [[ ${skip[*]} =~ "${resource}" ]]; then
                          ^-----------^ SC2076 (warning): Remove quotes from right-hand side of =~ to match as a regex rather than literally.


In gather_crs line 43:
      mkdir -p ${object_collection_path}
               ^-----------------------^ SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
      mkdir -p "${object_collection_path}"


In gather_crs line 44:
      /usr/bin/oc get ${resource} -n ${ocproject} -o yaml ${ocobject} &> ${object_collection_path}/${ocobject}.yaml & 
                      ^---------^ SC2086 (info): Double quote to prevent globbing and word splitting.
                                     ^----------^ SC2086 (info): Double quote to prevent globbing and word splitting.
                                                          ^---------^ SC2086 (info): Double quote to prevent globbing and word splitting.
                                                                         ^-----------------------^ SC2086 (info): Double quote to prevent globbing and word splitting.
                                                                                                   ^---------^ SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
      /usr/bin/oc get "${resource}" -n "${ocproject}" -o yaml "${ocobject}" &> "${object_collection_path}"/"${ocobject}".yaml & 


In gather_crs line 50:
for ns in ${namespaces[@]}; do
          ^--------------^ SC2068 (error): Double quote array expansions to avoid re-splitting elements.


In gather_crs line 51:
  for dpa in $(oc get dpa --namespace ${ns} --no-headers | awk '{print $1}'); do
                                      ^---^ SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
  for dpa in $(oc get dpa --namespace "${ns}" --no-headers | awk '{print $1}'); do


In gather_crs line 53:
    oc get dpa ${dpa} -o yaml --namespace ${ns} &> "/must-gather/clusters/${clusterID}/namespaces/${ns}/oadp.openshift.io/dpa-${dpa}/${dpa}.yml" &
               ^----^ SC2086 (info): Double quote to prevent globbing and word splitting.
                                          ^---^ SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
    oc get dpa "${dpa}" -o yaml --namespace "${ns}" &> "/must-gather/clusters/${clusterID}/namespaces/${ns}/oadp.openshift.io/dpa-${dpa}/${dpa}.yml" &

For more information:
  https://www.shellcheck.net/wiki/SC2068 -- Double quote array expansions to ...
  https://www.shellcheck.net/wiki/SC2145 -- Argument mixes string and array. ...
  https://www.shellcheck.net/wiki/SC2076 -- Remove quotes from right-hand sid...

```

Reply:
done